freecad-qt6: Qt6 version (with wayland per default)

ppenguin opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

A "parallel" package for FreeCAD built with Qt6 only, intended as "staging" to eventually supersede the Qt5 version permanently. ~~Could probably be~~ done on a shared code base (of `package.nix`)~~, but that would be more disruptive~~.

It appears to work well by just "naively" replacing/removing Qt5 libraries and build switches.

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

ppenguin commented:
Oh, it appears I have to follow this guideline, but it doesn't seem to work yet... https://github.com/NixOS/nixpkgs/tree/master/pkgs/by-name#recommendation-for-new-packages-with-multiple-versions.

Using `default.nix` or `package.nix` as a "wrapper" that does `callPackage` for the variants makes it not find the package at all...

ppenguin commented:
BTW, there seem to be many formatting changes even though I used `nixfmt-rfc-style` on it as required?!

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/5145

mmlb commented:
You should rebase to remove the reformats of all-packages.nix while doing that I would say to refactor the PR to remove the whole qt6 version in separate file from history all together. I would have a separate commit for just the reformatting and have it be the first commit. Lastly only the first commit message matches the commit message guidelines by having the package name as the prefix.

ppenguin commented:
@mmlb Squashed and rebased, should be good to go...

mmlb commented:
lgtm but you'll need some one to merge I don't have the perms to do so

ppenguin commented:
> lgtm but you'll need some one to merge I don't have the perms to do so

@wegank or @srounce maybe?

wegank commented:
> I would have a separate commit for just the reformatting and have it be the first commit. Lastly only the first commit message matches the commit message guidelines by having the package name as the prefix.

I'd suggest the same. By separating the changes into two commits, you will get a PR that is much easier to review. Also, the commit message isn't perfect, it should at least start with `freecad:` or `freecad-qt6:` instead of `freecad 1.0.0:`.

ppenguin commented:
> > I would have a separate commit for just the reformatting and have it be the first commit. Lastly only the first commit message matches the commit message guidelines by having the package name as the prefix.
> 
> I'd suggest the same. By separating the changes into two commits, you will get a PR that is much easier to review. Also, the commit message isn't perfect, it should at least start with `freecad:` or `freecad-qt6:` instead of `freecad 1.0.0:`.

I improved the commit message.
As for "reformatting", there is no isolated reformatting in this PR, so splitting is not easily possible; the derivation was significantly refactored to support the Qt5/Qt6 conditionals.

wegank committed:
freecad: reformat with nixfmt

ppenguin committed:
freecad-qt6: init (with wayland per default)

Includes following changes:

- remove duplicates with freecad (Qt5)
- refactor to use one derivation for both Qt5 and Qt6 builds
- add py-slvs (runtime dep for ondsel assy solver) -> will also fix
freecad (Qt5) upon merge

ppenguin commented:
@wegank Thanks for staying with this and finally merging. Cheers!
